scripts/get_intervals.py

```diff
@@ -98,129 +69,129 @@ def get_tiling_intervals(sizes, intervalsize, shift=None):
-        df_to_bed(train, out_file_path)
+        out_file_name = str(intervalsize) + '.training_intervals.bed'
+        train_file_path = os.path.join(out_dir, out_file_name)
+        df_to_bed(train, train_file_path)
 
         # Generate validation intervals - do not overlap
         _logger.info("Generating val intervals")
-        val_sizes = sizes[sizes['chrom'] == args.val]
+        val_sizes = sizes[sizes['chrom'] == val]
         val = get_tiling_intervals(
-            val_sizes, args.intervalsize)
+            val_sizes, intervalsize)
 
         # Write to file
-        if args.prefix is None:
-            out_file_name = 'val_intervals.bed'
-        else:
-            out_file_name = args.prefix + '.val_intervals.bed'
-        out_file_path = os.path.join(args.out_dir, out_file_name)
-        df_to_bed(val, out_file_path)
+        out_file_name = str(intervalsize) + '.val_intervals.bed'
+        val_file_path = os.path.join(out_dir, out_file_name)
+        df_to_bed(val, val_file_path)
 
         # Generate holdout intervals - do not overlap
-        if args.holdout is not None:
+        if holdout is not None:
             _logger.info("Generating holdout intervals")
-            holdout_sizes = sizes[sizes['chrom'] == args.holdout]
+            holdout_sizes = sizes[sizes['chrom'] == holdout]
             holdout = get_tiling_intervals(
-                holdout_sizes, args.intervalsize)
+                holdout_sizes, intervalsize)
 
             # Write to file
-            if args.prefix is None:
-                out_file_name = 'holdout_intervals.bed'
-            else:
-                out_file_name = args.prefix + '.holdout_intervals.bed'
-            out_file_path = os.path.join(args.out_dir, out_file_name)
-            df_to_bed(holdout, out_file_path)
-
-    _logger.info('Done!')
-
-
-if __name__ == "__main__":
-    main()
+            out_file_name = str(intervalsize) + '.holdout_intervals.bed'
+            holdout_file_path = os.path.join(out_dir, out_file_name)
+            df_to_bed(holdout, holdout_file_path)
+        _logger.info('Done!')
+        return train_file_path, val_file_path, holdout_file_path
+    elif regions is not None:
+        # If given regions is a file, then just return the file path
+        if regions.endswith(".bed"):
+            return regions
+        else:
+            final_intervals = pd.DataFrame()
+            regions = regions.strip("[]").split(",")
+            for region in regions:
+                chrom = region
+                chrom_range = None
+                # If regions are specified with intervals like chr1:0-50
+                # Then split the region into chrom and it's range.
+                if region.find(":") != -1:
+                    chrom, chrom_range = region.split(":")
+                chrom_sizes = sizes[sizes['chrom'] == chrom]
+                intervals = get_tiling_intervals(chrom_sizes, intervalsize)
+
+                if chrom_range:
```

Comment:
This behavior wouldn't denoise the regions asked for by the user. If, for example, intervalsize is 1000, and the user specifies "chr1:1500-2500"
- the intervals produced by L182 would be:
chr1 0 1000
chr1 1000 2000
chr1 2000 3000
- and none of these intervals would be selected by L186-189.

We should retain any intervals that intersect at all with the specified region. 
Should also explain this behavior in the help / docs.